### PR TITLE
Use SchedState to handle RFT configuration

### PR DESCRIPTION
**Pull request by joakim-hove**
Handle RFT with the SchedState mechanism

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3071 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3071 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3071 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3071 please

File: src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -663,39 +641,13 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
...
-            RFTConfig::PLT PLTKey = RFTConfig::PLTFromString(record.getItem("OUTPUT_PLT").getTrimmedString(0));
-            const auto well_names = wellNames(wellNamePattern, currentStep);
-            for (const auto& well_name : well_names) {
-                this->rft_config.updateRFT(well_name, currentStep, RFTKey);
-                this->rft_config.updatePLT(well_name, currentStep, PLTKey);
-            }
+    std::optional<std::size_t> Schedule::first_RFT() const {
+        for (std::size_t report_step = 0; report_step < this->snapshots.size(); report_step++) {
+            if (this->snapshots[report_step].rft_config().active())
+                return report_step;
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:648]: Just for the record this predicate used to be constant time.  Now it's linear in the number of report steps.

File: src/opm/output/eclipse/EclipseIO.cpp [hunk 2]
```diff
@@ -160,7 +159,7 @@ bool EclipseIO::Impl::wantRFTOutput( const int  report_step,
 {
     return !isSubstep
         && (static_cast<std::size_t>(report_step)
-            >= this->schedule.rftConfig().firstRFTOutput());
+            >= this->schedule.first_RFT());
```
Comment (bska) [inline src/opm/output/eclipse/EclipseIO.cpp:162]: I'm probably missing something, but
```C++
report_step >= std::optional<std::size_t>{}
```
is true for all values of `report_step` so this predicate is equivalent to
```C
return !isSubStep;
```
if `first_RFT()` returns a default-constructed `optional<std::size_t>` (i.e., a `std::nullopt`).  Is that really the intended behaviour of `first_RFT()`?  Does this get us back to the state where we always create empty `.RFT` files if the case does **not** request RFT output?

**Review by bska — COMMENTED** [review]
I notice a couple of potential issues with the new `first_RFT()` function.

File: src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:648]: Yes  you are right ;-)

File: src/opm/output/eclipse/EclipseIO.cpp (hunk 2 above)
Comment (joakim-hove) [reply src/opm/output/eclipse/EclipseIO.cpp:162]: Ehh - no; this was a bug - fix here: https://github.com/OPM/opm-common/pull/2308

**Pull request merged**